### PR TITLE
FEAT: Cart-management using mockbee Backend

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

Cart management is to manage the cart of products that the user wants to buy. It contains all the products along with the price receipt.

### 🚀 New behavior

- Add products into the cart and store them in DB using post request to API
- Remove products from the cart using delete request to API
- Increase the quantity of products in the cart using post request to API
- Price details are calculated accordingly based on price and discount.

### 🖼 Preview Link

[https://viskart-react-version-d1gpmnk0b-vishalpatil18.vercel.app/](https://viskart-react-version-d1gpmnk0b-vishalpatil18.vercel.app/)

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue #9 

> NOTE: It is not responsive yet. Will work on responsiveness at the end.

### CODE REVIEW

**Review by Kevin-Solomon — APPROVED** [review]
- Clean UI
- Nice abstractions

File: src/context/authModalContext.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      switch (action) {
+        case "OPEN":
+          return { login: true, signup: false };
+        case "CLOSE":
+          return { login: false, signup: false };
+        case "SIGNUP":
+          return { login: false, signup: true };
+        case "LOGIN":
+          return { login: true, signup: false };
+        default:
```
Comment (starver20) [inline src/context/authModalContext.js:25]: OPEN and LOGIN cases are returning same state, you can try ti club them together

File: src/context/authModalContext.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    setAuthModalOpen(() => {
+      switch (action) {
+        case "OPEN":
+          return { login: true, signup: false };
+        case "CLOSE":
+          return { login: false, signup: false };
+        case "SIGNUP":
+          return { login: false, signup: true };
+        case "LOGIN":
+          return { login: true, signup: false };
```
Comment (starver20) [inline src/context/authModalContext.js:24]: OPEN and LOGIN cases are returning same state, you can try ti club them together

File: src/context/cartContext.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import { initialCartState, cartReducer } from "../redux";
```
Comment (starver20) [inline src/context/cartContext.js:2]: We are'nt actually using redux, so the name does not seem to be correct

File: src/context/cartContext.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  const { authState } = useAuth();
+
+  useEffect(() => {
+    authState.token
+      ? (async () => {
+          try {
+            console.log("Authstate token in cartcontext", authState.token);
+            console.log("Useeffect called initially");
+            const response = await cartService(authState.token);
+            console.log(response.data.cart);
```
Comment (starver20) [inline src/context/cartContext.js]: Can remove the console.log's

File: src/context/cartContext.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            cartDispatch({
+              type: "INITIAL_DATA",
+              payload: response.data.cart,
+            });
+          } catch (error) {
+            console.log("ERROR: ", error);
+          }
+        })()
+      : console.log("Useeffect called initially but authstate.user is false");
```
Comment (starver20) [inline src/context/cartContext.js]: Can show error instead

File: src/services/removeFromCart.service.js [hunk 6]
```diff
@@ -1,15 +1,9 @@
...
-    if (curr._id === item._id) {
-      if (item.quantity > 1)
-        acc = [...acc, { ...curr, quantity: curr.quantity - 1 }];
-    } else acc = [...acc, curr];
-    return acc;
-  }, []);
+const removeFromCartService = (item, token) => {
+  return axios.delete(`/api/user/cart/${item._id}`, {
+    headers: { authorization: token },
+  });
```
Comment (starver20) [inline src/services/removeFromCart.service.js:6]: Liked that you are creating seperate functions from the very start

**Review by starver20 — APPROVED** [review]
Code looks all good to merge. Well written code @VishalPatil18

File: src/context/authModalContext.js (hunk 2 above)
Comment (VishalPatil18) [reply src/context/authModalContext.js:24]: Yes, true but suppose in the future we need the modal for something else as well then we can directly use it by just using open and use login for only the login screen there...
This was my thought process.

File: src/context/cartContext.js (hunk 3 above)
Comment (VishalPatil18) [reply src/context/cartContext.js:2]: Yes, but the work is similar to redux right? Like I followed this folder structure which was shared by an MA and hence that naming.

Screenshot 2022-03-10 at 11 37 15

File: src/context/cartContext.js (hunk 4 above)
Comment (VishalPatil18) [reply src/context/cartContext.js]: sure will do it. Maybe missed those somehow😬

File: src/context/cartContext.js (hunk 5 above)
Comment (VishalPatil18) [reply src/context/cartContext.js]: Yes would change this...

File: src/context/cartContext.js (hunk 3 above)
Comment (starver20) [reply src/context/cartContext.js:2]: Oh okay, i wasnt aware of that, thanks

File: src/context/authModalContext.js (hunk 2 above)
Comment (starver20) [reply src/context/authModalContext.js:24]: Ya that makes sense

**Pull request merged**